### PR TITLE
Ft/upgrade build base to 1.22.1

### PR DESCRIPTION
**Pull request by Sotatek-TinnNguyen**
## Proposed changes

- Upgrade golang version from 1.20.6 to 1.22.1
- Upgrade image Redis's circleCI image and orb to new version

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,31 +15,31 @@ aliases:
...
+  codecov: codecov/codecov@4.0.1
+  docker: circleci/docker@2.5.0
 
 executors:
   linux-executor: # this executor is for general testing and packaging linux binaries
     working_directory: ~/go/src/github.com/klaytn/klaytn
     resource_class: medium
     docker:
-      - image: klaytn/build_base:1.2-ubuntu18.04-go.1.20.6-solc0.8.13
+      - image: klaytn/build_base:1.12-ubuntu20.04-go.1.20.6-solc0.8.13
```
Comment (yoomee1313) [inline .circleci/config.yml]: @Sotatek-TinnNguyen Have you tested in ubuntu18.04? I thought we need to support old ubuntu os.

Comment (yoomee1313) [inline .circleci/config.yml]: Name of docker image go version is go1.20.6. Shouldn't it be 1.22.1?

**Review by yoomee1313 — COMMENTED** [review]
Tested in various enviroment about next commands and it works correctly: macos, ubuntu20.04, ubunu22.04, centos
- ./ken version
- ./homi setup --gen-type=local --baobab-test --cn-num 1 --test-num 2
- ./kcn init --datadir data homi/scripts/genesis.json
- ./ken --datadir data

**Comment by yoomee1313** [comment]
@Sotatek-TinnNguyen Can you update the minimum go version of klaytn repository? We need to update it.
https://github.com/klaytn/klaytn/blob/5da72febaa0b6335f501c0bb86b94c07688ac717/go.mod#L3

File: .circleci/config.yml (hunk 1 above)
Comment (Sotatek-TinnNguyen) [reply .circleci/config.yml]: Okay, I will create another version for ubuntu 18.04

Comment (Sotatek-TinnNguyen) [reply .circleci/config.yml]: yes, maybe I missed it, I will fix now

Comment (yoomee1313) [reply .circleci/config.yml]: Thank you. By the way, if the binary works in ubuntu 18.04, no need to fix it. Do you know anything about this?

**Review by yoomee1313 — APPROVED** [review]

**Comment by yoomee1313** [comment]
@Sotatek-TinnNguyen I'll make a new PR for go minimum version upgrade. Could you open this PR and merge it?

**Review by blukat29 — APPROVED** [review]

**Pull request merged**